starknet_py/proxy_check.py

```diff
@@ -2,51 +2,72 @@
 # pyright: reportUndefinedVariable=false
 
 from abc import ABC, abstractmethod
```

Comment:
Since we are introducing breaking changes anyway we should move this and resolver to proper directory within starknet_py